### PR TITLE
awaitable full clean

### PR DESCRIPTION
**Pull request by troepolik**
Ability to call awaitable Full clean. To be sure it was cleared.
Could you be so kind to approve it or give me feedback what can I fix in it to merge?

### CODE REVIEW

File: src/FastCache.Cached/CacheManager.cs [hunk 1]
```diff
@@ -26,25 +26,27 @@ public static class CacheManager
...
     public static void QueueFullClear<K, V>() where K : notnull
     {
-        Task.Run(async static () =>
-        {
-            var evictionJob = CacheStaticHolder<K, V>.EvictionJob;
-            await evictionJob.FullEvictionLock.WaitAsync();
+        Task.Run(static () => FullCleanAsync<K, V>());
+    }
+
+    public static async Task FullCleanAsync<K, V>() where K : notnull
```
Comment (neon-sunset) [inline src/FastCache.Cached/CacheManager.cs:32]:
```suggestion
    /// <summary>
    /// Remove all cache entries of type Cached[K, V] from the cache
    /// </summary>
    /// <returns>A task that completes upon full clear execution.</returns>
    public static async Task ExecuteFullClear<K, V>() where K : notnull
```

**Review by neon-sunset — CHANGES_REQUESTED** [review]

**Comment by neon-sunset** [comment]
It does make sense to expose a task to wait clear completion in place. However, you will have to add public `ExecuteFullEviction` too.

**Comment by troepolik** [comment]
Done. But for me is not so clear how to properly run eviction. Please, look at changes closer.
Is I should call ImmediateFullEviction and is IsActive flag checked  properly?

**Comment by neon-sunset** [comment]
It will require additional changes to forward either the task of `Task.Run(ImmediateFullEviction<K, V>)` or `StaggeredFullEviction<K, V>`.. However, thinking about it some more, it can probably be handled in a more graceful way by "joining" an already running full eviction if any instead of throwing/blocking and re-running/not doing anything.

For now, just remove the `ExecuteFullEviction` part - I will commit it separately.

**Comment by troepolik** [comment]
Done

**Pull request merged**